plugins/manager/lib/yform/manager/dataset.php

```diff
@@ -550,7 +550,14 @@ public function executeForm(rex_yform $yform, callable $afterFieldsExecuted = nu
         $exists = $this->exists();
         $oldData = $this->getData();
 
-        if ($exists) {
+        if (isset($yform->objparams['copied_from'])) {
+            $copy = clone($this);
```

Comment:
Ich finde nicht gut, dass `executeForm` innerhalb der ursprünglichen Instanz abläuft.
Also wer diese Methode, in der wir hier sind, in einer Modelklasse erweitert, der könnte sich dann wundern, dass executeForm für den ursprünglichen Datensatz ausgeführt wird.

Ich fände eigentlich besser, wenn der clone schon irgendwie vorher erzeugt werden würde, und darin dann die ganzen Methoden aufgerufen werden würden (also eigentlich auch `getForm` schon).

Wie gut das allerdings machbar ist, kann ich so aus dem Stand nicht sagen, dafür stecke ich gerade zu wenig drin in den Abläufen.

Reply:
> Wie gut das allerdings machbar ist, kann ich so aus dem Stand nicht sagen, dafür stecke ich gerade zu wenig drin in den Abläufen.

Das geht mir ebenso … Hätte es auch gern anders gelöst. Für mich ist es aber erst einmal die Stelle um den ExtensionPoint korrekt aufzurufen und die Daten zu beeinflussen.

Genaueres könnte @dergel sagen.

Reply:
Also ab hier hätte ich eigentlich schon gerne einen Clone statt den Original-Datensatz. Und dieser Clone sollte eigentlich schon als neuer Datensatz erkennbar sein, also ohne ID und exists=false.
https://github.com/yakamara/redaxo_yform/blob/5cd8deaa430643a252399bf13f1d214a33840734/plugins/manager/lib/yform/manager.php#L283